src/Xamarin.Android.Build.Tasks/Tests/Xamarin.Android.Build.Tests/Utilities/DeviceTest.cs

```diff
@@ -1,23 +1,33 @@
 using NUnit.Framework;
+using NUnit.Framework.Interfaces;
 using System;
-using System.Collections;
-using System.Collections.Generic;
 using System.Diagnostics;
 using System.IO;
-using System.Linq;
-using System.Security.Cryptography;
-using System.Text;
 using System.Text.RegularExpressions;
 using System.Threading;
-using System.Xml;
 using System.Xml.Linq;
 using System.Xml.XPath;
-using Xamarin.ProjectTools;
 
 namespace Xamarin.Android.Build.Tests
 {
 	public class DeviceTest: BaseTest
 	{
+		[TearDown]
+		protected override void CleanupTest ()
+		{
+			if (HasDevices && TestContext.CurrentContext.Result.Outcome.Status == TestStatus.Failed &&
+					TestOutputDirectories.TryGetValue (TestContext.CurrentContext.Test.ID, out string outputDir)) {
+				string local = Path.Combine (outputDir, "screenshot.png");
+				string remote = "/sdcard/screenshot.png";
```

Comment:
I'm not certain on the "reliability" of `/sdcard`; iirc the path of the SD Card mount point has changed over time, and have little reason to doubt that it won't change in the future.

Should this instead use `/data/local/tmp`?  We've been using that directory for *years* as part of Fast Deployment, so it *hasn't* changed.